opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -701,16 +701,30 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
     void Schedule::shut_well(const std::string& well_name, std::size_t report_step) {
         this->internalWELLSTATUSACTIONXFromPYACTION(well_name, report_step, "SHUT");
     }
+    void Schedule::shut_well(const std::string& well_name) {
+        this->internalWELLSTATUSACTIONXFromPYACTION(well_name, this->current_report_step, "SHUT");
+    }
 
     void Schedule::open_well(const std::string& well_name, std::size_t report_step) {
         this->internalWELLSTATUSACTIONXFromPYACTION(well_name, report_step, "OPEN");
     }
+    void Schedule::open_well(const std::string& well_name) {
+        this->internalWELLSTATUSACTIONXFromPYACTION(well_name, this->current_report_step, "OPEN");
+    }
 
     void Schedule::stop_well(const std::string& well_name, std::size_t report_step) {
         this->internalWELLSTATUSACTIONXFromPYACTION(well_name, report_step, "STOP");
     }
+    void Schedule::stop_well(const std::string& well_name) {
+        this->internalWELLSTATUSACTIONXFromPYACTION(well_name, this->current_report_step, "STOP");
+    }
 
     void Schedule::internalWELLSTATUSACTIONXFromPYACTION(const std::string& well_name, std::size_t report_step, const std::string& wellStatus) {
+        if (report_step < this->current_report_step) {
+            throw std::invalid_argument("Well status change for past report step " + std::to_string(report_step) + " requested, current report step is " + std::to_string(this->current_report_step) + ".");
+        } else if (report_step >= this->m_sched_deck.size()) {
+            throw std::invalid_argument("Well status change for report step " + std::to_string(report_step) + " requested, this exceeds the total number of report steps, being " + std::to_string(this->m_sched_deck.size() -1) + ".");
+        }
```

Comment:
We already `#include` [`<fmt/format.h>`](https://github.com/OPM/opm-common/blob/e3276a36c76ff7d5e413ac13f87f660c823125f0/opm/input/eclipse/Schedule/Schedule.cpp#L110) in this file, so I think I'd prefer to use that over `std::to_string()` here.